app/src/main/java/com/example/atheneum/fragments/AddBookFragment.java

```diff
@@ -27,11 +54,129 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         this.view = inflater.inflate(R.layout.fragment_add_edit_book, container, false);
 
         this.context = getContext();
+
+        // bind editText references to UI elements
+        titleEditText = this.view.findViewById(R.id.bookTitleEditText);
+        authorEditText = this.view.findViewById(R.id.authorEditText);
+        isbnEditText = this.view.findViewById(R.id.isbnEditText);
+        descEditText = this.view.findViewById(R.id.descEditText);
+
+        saveBtn = this.view.findViewById(R.id.saveBookBtn);
+        saveBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                saveNewBook();
+            }
+        });
+        scanIsbnBtn = this.view.findViewById(R.id.scanIsbnButton);
+        scanIsbnBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                scanIsbn();
+            }
+        });
+        autoPopulateByIsgnBtn = this.view.findViewById(R.id.populateFromIsbnBtn);
+        autoPopulateByIsgnBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                populateFieldsByIsbn();
+            }
+        });
+
         mainActivity = (MainActivity) getActivity();
         // set action bar title
         mainActivity.setActionBarTitle(context.getResources().getString(R.string.add_book_prompt));
 
         return this.view;
     }
 
+    public void scanIsbn() {
+        Log.i("AddBook", "AddBook*** ISBN scan requested");
+        // TODO
+    }
+
+    public void populateFieldsByIsbn() {
+        // TODO: FINISH
+
+        Log.i("AddBook", "AddBook*** Auto-populate requested");
+        long isbn = -1;
+        if (!isbnEditText.getText().toString().equals("")) {
```

Comment:
You could just use TextUtils.isEmpty() if checking if an EditText is empty or not. I don't expect you to have to change this since it's minor but just an FYI. https://developer.android.com/reference/android/text/TextUtils.html#isEmpty(java.lang.CharSequence)

Reply:
Yeah that's definitely cleaner, fixed.